### PR TITLE
Intly 2114 move upgrades to product roles

### PR DESCRIPTION
**Pull request by sedroche**
## Additional Information
Jira: https://issues.jboss.org/browse/INTLY-2114

Currently there is logic in the upgrade playbooks that belongs in the product roles. This PR refactors this logic into the correct roles

## Verification Steps

1. Install integreatly `v1.3`
2. Create the namespace `openshift-integreatly-backups`
3. Create a secret called `s3-credentials` in `openshift-integreatly-backups`
```bash
    oc create secret generic s3-credentials -n openshift-integreatly-backups  --dry-run -o yaml | oc apply -f -
```
4. Run the backup playbook
```bash
ansible-playbook -i inventories/hosts playbooks/backup_restore/install.yml
```
5. Run the upgrade playbook in this branch
```bash
git checkout INTLY-2114-move-upgrades-to-product-roles
ansible-playbook -i inventories/hosts playbooks/upgrades/upgrade.yaml
```
## Is an upgrade task required and are there additional steps needed to test this?
No

### CODE REVIEW

**Comment by odra** [comment]
:eyes:

File: playbooks/upgrades/upgrade.yaml [hunk 1]
```diff
@@ -2,81 +2,36 @@
...
-    - name: Retrieve SSO host
-      shell: oc get route sso -n {{ eval_rhsso_namespace }} -o jsonpath='{.spec.host}'
-      register: get_sso_host
-    - set_fact:
-        upgrade_sso_url: "https://{{ get_sso_host.stdout | trim }}/auth/admin/openshift/console"
+    - include_role:
+        name: rhsso
+        tasks_from: upgrade
+      vars:
+        upgrade_sso_operator_image: quay.io/integreatly/keycloak-operator:{{ rhsso_operator_release_tag }}
```
Comment (odra) [inline playbooks/upgrades/upgrade.yaml]: I know the image name was already hardcoded but are there variables in the manifest file that specify the image name as well?

Comment (sedroche) [reply playbooks/upgrades/upgrade.yaml]: It isn't there currently.

https://github.com/integr8ly/installation/blob/398e4cb7bfb4c4ed777fc1b904cfb45a2da2c526/inventories/group_vars/all/manifest.yaml#L62-L67

I can add it if you want?

Comment (odra) [reply playbooks/upgrades/upgrade.yaml]: Nah it is fine, we can do it in another pr if needed.

Just aiting for my cluster to be created so I can give it a try.

**Comment by odra** [comment]
Got this error:

```
TASK [Set eval_app_host var] **************************************************************************************************************************************************************************************
fatal: [127.0.0.1]: FAILED! => {"msg": "The task includes an option with an undefined variable. The error was: \"hostvars['EVAL_VARS']\" is undefined\n\nThe error appears to have been in '/root/installation/playbooks/generate-customisation-inventory.yml': line 15, column 7, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n    - name: Set eval_app_host var\n      ^ here\n"}

```

**Comment by sedroche** [comment]
@odra I have updated with the changes to run the master node tasks when required.

https://github.com/integr8ly/installation/compare/92d3ce5cb03be7236af890ff52eabbda8ec57c09..1ec2ab9e365f7d991bf0ff2bb397bb3b516bb550

**Comment by odra** [comment]
ok will give it a try

**Comment by odra** [comment]
Got this error now:
```
TASK [openshift : Retrieve master configurations] *****************************************************************************************************************************************************************
fatal: [127.0.0.1]: FAILED! => {"changed": false, "msg": "file not found: /etc/origin/master/master-config.yaml"}
```

**Comment by sedroche** [comment]
These changes are just for 1.4.
Closing for https://github.com/integr8ly/installation/pull/667

**Pull request closed without merging**